theano/sparse/sandbox/sp2.py

```diff
@@ -132,7 +136,15 @@ def perform(self, node, (n, p, shape, ), (out, )):
         out[0] = csx_matrix(binomial, dtype=self.dtype)
 
     def grad(self, (n, p, shape, ), (gz,)):
-        return None, None, None
+        comment_n = "No gradient exists for the number of samples in class\
+                     Binomial of theano/sparse/sandbox/sp2.py"
+        comment_p = "No gradient exists for the prob of success in class\
+                     Binomial of theano/sparse/sandbox/sp2.py"
+        return [theano.gradient.grad_undefined(op=self, x_pos=0, x=n,
+                                               comment=comment_n),
+                theano.gradient.grad_undefined(op=self, x_pos=1, x=p,
+                                               comment=comment_p),
+                theano.gradient.disconnected_type()]
```

Comment:
grad_undefined means the gradient cannot be defined since there is no gradient for such cases. It's much better to have it and tell the user that you cannot get the gradient rather than having nothing.